storage/client_test.go

```diff
@@ -337,16 +337,21 @@ func TestUpdateRangeAddressing(t *testing.T) {
 	for i, test := range testCases {
 		left := &proto.RangeDescriptor{RaftID: int64(i * 2), StartKey: test.leftStart, EndKey: test.leftEnd}
 		right := &proto.RangeDescriptor{RaftID: int64(i*2 + 1), StartKey: test.rightStart, EndKey: test.rightEnd}
+		var calls []*client.Call
 		if test.split {
-			if err := storage.SplitRangeAddressing(store.DB(), left, right); err != nil {
+			var err error
+			if calls, err = storage.SplitRangeAddressing(left, right); err != nil {
```

Comment:
Ah...now I see why you created the client.Txn object as separate...these tricky cases where we just want to pass a client down through an execution stack and use Prepare instead of accumulating the calls to be passed back up the execution stack.

I think this might be a step backwards. Hate to suggest it because I know this was a huge amount of drudgery, but why don't we limit this change to moving everything to client.Call? The previous incarnation looked a lot better for cases like this and others you've been experiencing, where code (like range addressing or the range spanning tree) could be agnostic about whether the client.KV object it was using was transactional or not. I think we should keep Run() as variadic instead of Call() but should move the functionality now in Txn back into KV. Keeping a slice of Call objects is a bigger pain in the ass than we thought. Also, having Txn and KV be different types is ugly.